src/routes/organization/[organizationId]/+page.server.ts

```diff
@@ -0,0 +1,16 @@
+import { authenticate, checkSessionAuth, getUsers, readPath, verifySessionCookie } from '$lib/server/firebaseUtils.js';
+import { error, redirect } from '@sveltejs/kit';
+import type { PageServerLoad } from './$types.js';
+
+export const load = (async ({ cookies, params: { organizationId }, url }) => {
+  const organization: Database.Organization = await readPath(`/organizations/${organizationId}`);
+  if (!organization) throw error(404);
+  const { members = {} } = organization.private ?? {};
+  await checkSessionAuth(cookies, {
+    loginRedirect: url.pathname,
+    authFunction: async ({ uid }) => (members[uid]?.role === 'admin')
+  });
```

Comment:
An example of how easy it is to authenticate server-side. If all we wanted to do was check that they are logged in, we wouldn't even need the options (except maybe the `loginRedirect` unless we want them to be redirected home after login)

Reply:
very cool!